apps-rendering/src/renderer.ts

```diff
@@ -214,8 +230,11 @@ const textElement =
 			textElement(format, isEditions),
 		);
 		switch (node.nodeName) {
-			case 'P':
-				return h(Paragraph, { key, format }, children);
+			case 'P': {
+				const showDropCap =
+					allowsDropCaps(format) && text.length >= 200;
```

Comment:
This is true.

Also, can `node.textContent` contain html, like an a tag? We remove these in DCR before counting the number of characters

Reply:
> I believe the first word also has to have a minimum length of 3 letters.

I couldn't find this particular rule in DCR's `TextBlockComponent.tsx`, which is why I didn't implement in AR... see 
https://www.theguardian.com/artanddesign/2022/apr/19/savage-unstoppable-painter-everlyn-nicodemus-kilimanjaro-defiance-richard-saltoun 
The first word is 2 characters long and the drop cap is still shown 🧐

> Also, can node.textContent contain html, like an a tag? We remove these in DCR before counting the number of characters

Good question - I checked this during development and `node.textContent` is pure text. For instance, console logging that above article's first paragraph:

<img width="1008" alt="image" src="https://user-images.githubusercontent.com/57295823/164281370-a245c572-7cfd-48bc-ac54-a88a5285f89a.png">

Reply:
> I couldn't find this particular rule in DCR

Ahem, yes. I suspect we just haven't implemented it yet 😊 

I defo recall agreeing to this rule though

>  I checked this during development and node.textContent is pure text

Great!

Reply:
> we just haven't implemented it yet  

I see. I'm thinking about how to implement this, and the naive solution, i.e. checking for the length of `text.split(" ")[0]` seems a bit inefficient because we'd be creating an array of strings in memory for every single paragraph...
a regex sitting outside the rendering function - so it doesn't get recompiled every time - seems a viable alternative but taking into account all the caveats such as the quotation marks makes me think it's not the easiest regex to write.

Would you mind if we took this to a separate PR? I'll open an issue to address this extra rule, while bringing parity with DCR in this one.

Reply:
`!String.slice(0,3).includes(" ")` ?

Reply:
@MarSavar Here are the dropcap colours for dark mode:
news: news.500 (#FF5943)
opinion: opinion.500 (#FF7F0F)
sport: sport.500 (#00B2FF)
culture: culture.400 (#A1845C)
lifestyle: lifestyle.450 (#F37ABC)

Reply:
> @MarSavar Here are the dropcap colours for dark mode:
> news: news.500 (#FF5943)
> opinion: opinion.500 (#FF7F0F)
> sport: sport.500 (#00B2FF)
> culture: culture.400 (#A1845C)
> lifestyle: lifestyle.450 (#F37ABC)

Thanks @benwuersching - lifestyle.450 doesn't seem to exist in [source foundations](https://guardian.github.io/source/?path=/story/packages-source-foundations-palette--page), the only options are 400 and 500. I could hardcode it in the palette but would prefer using source if possible. Is there an alternative? I am using `lifestyle[500]` in this PR.

@mxdvl @oliverlloyd Re: the three character rule for dropcaps (4 if including the initial quotation mark) -> I have just pushed a commit that deals with all the edge cases with a regex. I have benchmarked it against string slicing (1 million iterations) and the performance is equivalent or better (not that it matters that much considering we cache content but I prefer a one liner solution to a function that deals with lots of edge cases). I've also added a JSdoc comment to explain the regex. Hope it's helpful

Reply:
@MarSavar There is a separate set called Colour (Dark mode) which has lifestyle.450. It exists in Figma but doesn't seem to have made it into the assets you can use. How disappointing... For the time being, just use lifestyle.500. Thank you.

Reply:
@maxdvl @oliverlloyd 

For reference here is the benchmark:

```javascript
const dropCapRegex = /^["'\u2018\u201c]?[a-zA-Z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u024F]{3,}/;
const stringToCheck = "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.";

const checkLength = (text) => {
    const quotationMarks = ["'", "\"", "\u2018", "\u201c"];
    if (quotationMarks.some(quotationMark => text.startsWith(quotationMark))) {
        return !text.slice(0,4).includes(" ");
    }
    return !text.slice(0,3).includes(" ");
}

// Regex
console.time();
for (let i=0; i<1000000; i++) {
    const shouldShow = dropCapRegex.test(stringToCheck);
}
console.timeEnd();


// String manipulation
console.time();
for (let i=0; i<1000000; i++) {
    const shouldShow = checkLength(stringToCheck);
}
console.timeEnd(); 
```

Results:

Regex: `32.388916015625 ms`
String manipulation with edge cases: `80.099853515625 ms`

So it turns out that with longer paragraphs the regex is nearly 3 times faster!
Sorry for derailing the discussion but hopefully this is useful for a DCR implementation.